### PR TITLE
Adds `CODEOWNERS` file to streamline notifications

### PR DESCRIPTION
**Pull request by jsmith-bdai**
# Description

This PR adds in a [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) file so that we can better handle the review request process.

NOTE: This version is not finalized, please add suggestions to other areas on this PR. We can constantly add more as well, I just wanted to get an initial version up and filled in the things I knew off the top of my head :smile:  

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Comment by pascal-roth** [comment]
Looks like a cool feature to me! I would personally do not include teams as there might be a lot of changes within the people. But that can be up for discussion

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Teams
+# These don't currently work - need to figure out how to do team codeowners
+# @team-aii-devs @hhansen-bdai @jsmith-bdai @jtigue-bdai
+# @team-nvidia @Dhoeller19 @kellyguo11 @Mayankm96 @nikitardn # TODO: Add other members
+# @team-rsl @Mayankm96 @nikitardn @pascal-roth # TODO: Add other members
```
Comment (jsmith-bdai) [inline .github/CODEOWNERS]: Leaving this here in case we want to remove teams section
```suggestion
```

**Comment by jsmith-bdai** [comment]
> Looks like a cool feature to me! I would personally do not include teams as there might be a lot of changes within the people. But that can be up for discussion

@pascal-roth Thanks! Please add any section of the code for which you would like to be a reviewer for as suggestions

**Comment by Mayankm96** [comment]
@jsmith-bdai I took a stab at having a more thorough listing in there. Still a pretty tentative list. We can work on improving it in our discussions.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+/source/extensions/omni.isaac.lab/omni/isaac/lab/managers @jsmith-bdai @Dhoeller19 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors @jsmith-bdai @Dhoeller19 @pascal-roth @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/camera @kellyguo11 @pascal-roth
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/contact_sensor @jtigue-bdai
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/frame_transformer @jsmith-bdai
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster @pascal-roth @Dhoeller19
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sim @Mayankm96 @jsmith-bdai
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/simulation_context.py @Dhoeller19 @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/terrains @Dhoeller19 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/utils @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96
```
Comment (jtigue-bdai) [inline .github/CODEOWNERS]: Do we want to break out the sub-folders of utils?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+/source/extensions/omni.isaac.lab/omni/isaac/lab/actuators @Dhoeller19 @Mayankm96 @nikitardn
+/source/extensions/omni.isaac.lab/omni/isaac/lab/app @hhansen-bdai @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/assets @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/assets/deformable_object @kellyguo11 @Mayankm96 @masoudmoghani
+/source/extensions/omni.isaac.lab/omni/isaac/lab/controllers @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/envs @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_* @jsmith-bdai @Dhoeller19 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_* @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/managers @jsmith-bdai @Dhoeller19 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors @jsmith-bdai @Dhoeller19 @pascal-roth @Mayankm96
```
Comment (jtigue-bdai) [inline .github/CODEOWNERS]:
```suggestion
/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors @jsmith-bdai @Dhoeller19 @pascal-roth @Mayankm96 @jtigue-bdai
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (jtigue-bdai) [inline .github/CODEOWNERS]:
```suggestion
/source/extensions/omni.isaac.lab/omni/isaac/lab/utils @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96 @jtigue-bdai 
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+# @team-nvidia @Dhoeller19 @kellyguo11 @Mayankm96 @nikitardn # TODO: Add other members
+# @team-rsl @Mayankm96 @nikitardn @pascal-roth # TODO: Add other members
+
+# App experience files
+# These are the files that are used to launch the app with the correct settings and configurations
+/source/apps/ @kellyguo11 @hhansen-bdai @Mayankm96 @Dhoeller19
+
+# Core Framework
+/source/extensions/omni.isaac.lab/ @Dhoeller19 @Mayankm96 @jsmith-bdai @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/actuators @Dhoeller19 @Mayankm96 @nikitardn
```
Comment (jtigue-bdai) [inline .github/CODEOWNERS]:
```suggestion
/source/extensions/omni.isaac.lab/omni/isaac/lab/actuators @Dhoeller19 @Mayankm96 @nikitardn @jtigue-bdai 
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster @pascal-roth @Dhoeller19
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sim @Mayankm96 @jsmith-bdai
+/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/simulation_context.py @Dhoeller19 @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/terrains @Dhoeller19 @Mayankm96
+/source/extensions/omni.isaac.lab/omni/isaac/lab/utils @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96
+
+# RL Environment
+/source/extensions/omni.isaac.lab_tasks/ @Dhoeller19 @Mayankm96 @jsmith-bdai @kellyguo11
+/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct @Dhoeller19 @kellyguo11
+/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based @Dhoeller19 @Mayankm96 @jsmith-bdai
```
Comment (jtigue-bdai) [inline .github/CODEOWNERS]:
```suggestion
/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based @Dhoeller19 @Mayankm96 @jsmith-bdai @jtigue-bdai 
```

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+# App experience files
+# These are the files that are used to launch the app with the correct settings and configurations
+/source/apps/ @kellyguo11 @hhansen-bdai @Mayankm96 @Dhoeller19
+
+# Core Framework
+/source/extensions/omni.isaac.lab/ @Dhoeller19 @Mayankm96 @jsmith-bdai @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/actuators @Dhoeller19 @Mayankm96 @nikitardn
+/source/extensions/omni.isaac.lab/omni/isaac/lab/app @hhansen-bdai @kellyguo11
+/source/extensions/omni.isaac.lab/omni/isaac/lab/assets @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96
```
Comment (jtigue-bdai) [inline .github/CODEOWNERS]:
```suggestion
/source/extensions/omni.isaac.lab/omni/isaac/lab/assets @jsmith-bdai @Dhoeller19 @kellyguo11 @Mayankm96 @jtigue-bdai 
```

**Review by jtigue-bdai — APPROVED** [review]
Added suggestions for some areas of interest for me.

**Review by hhansen-bdai — APPROVED** [review]

**Review by kellyguo11 — APPROVED** [review]

**Pull request merged**